prog/test/postgres_resource.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Test::PostgresResource < Prog::Test::Base
+  semaphore :destroy
+
+  def self.assemble
+    postgres_service_project = Project.create(name: "Postgres-Service-Project") { _1.id = Config.postgres_service_project_id }
+    postgres_test_project = Project.create(name: "Postgres-Test-Project")
+
+    Strand.create_with_id(
+      prog: "Test::PostgresResource",
+      label: "start",
+      stack: [{
+        "postgres_service_project_id" => postgres_service_project.id,
+        "postgres_test_project_id" => postgres_test_project.id
+      }]
+    )
+  end
+
+  label def start
+    st = Prog::Postgres::PostgresResourceNexus.assemble(
+      project_id: frame["postgres_test_project_id"],
+      location: "hetzner-fsn1",
+      name: "postgres-test-standard",
+      target_vm_size: "standard-2",
+      target_storage_size_gib: 128
+    )
+
+    update_stack({"postgres_resource_id" => st.id})
+    hop_wait_postgres_resource
+  end
+
+  label def wait_postgres_resource
+    if postgres_resource.strand.label == "wait" &&
+        postgres_server.run_query("SELECT 1") == "1"
+      hop_test_postgres
+    else
+      nap 10
+    end
+  end
+
+  label def test_postgres
+    unless postgres_server.run_query(test_queries_sql) == "DROP TABLE\nCREATE TABLE\nINSERT 0 10\n4159.90\n415.99\n4.1"
+      update_stack({"fail_message" => "Failed to run test queries"})
+    end
+
+    hop_destroy_postgres
+  end
+
+  label def destroy_postgres
+    postgres_resource.incr_destroy
+    hop_destroy
+  end
+
+  label def destroy
+    postgres_service_project.destroy
+    postgres_test_project.destroy
+
+    frame["fail_message"] ? fail_test(frame["fail_message"]) : hop_finish
+  end
+
+  label def finish
```

Comment:
No need for separate label for this, you can directly pop in the previous label.